public/scripts/project/lock.js

```diff
@@ -0,0 +1,75 @@
+define(function(require) {
```

Comment:
You need a comment block at the top of this file explaining the lock philosophy and proper use.